chaoscenter/web/src/views/ChaosStudio/ChaosStudio.tsx

```diff
@@ -160,26 +158,29 @@ export default function ChaosStudioView({
       const probeWithoutAnnotation = await (
         experimentHandler as KubernetesYamlService
       )?.checkProbesInExperimentManifest(experiment?.manifest as KubernetesExperimentManifest);
-      probeWithoutRef.current = probeWithoutAnnotation;
 
-      /**
-       * Checks if probe metadata is already present in the manifest
-       *
-       * Returns true if probe metadata is present
-       */
-      const doesProbeExists = await (experimentHandler as KubernetesYamlService)?.doesProbeMetadataExists(
-        experiment?.manifest as KubernetesExperimentManifest
-      );
+      if (probeWithoutAnnotation) {
```

Comment:
This check conflicts with the logic we are trying to enforce. `probeWithoutAnnoitation` returns the name of the fault which doesn't have the `probeRef` annotation or the new resilience probes in the manifest basically. With the above logic it checks if `probeWithoutAnnotation` -> returns a string with the fault name if res probes aren't present, then it goes inside the check and checks `doesProbeExists` which are basically checking if legacy probes with inline support are present in the manifest or not. If it is present, then it returns true. The next check verifies `if (!true)` this is a case where there is no res probes but an old probe, it bypasses the warning because of `if (!true)` whereas it should be shown since res probes are missing.

The previous logic used the `if (probeWithoutAnnotation)` to validate if res probe are present or not

Reply:
Here's a flow @S-ayanide , As you can see `ChaosExperimentService` directly calls `ProbeService(handler)` instead of calling graphql mutation.
![image](https://github.com/litmuschaos/litmus/assets/53862866/66c4d6f8-e7e6-4058-9296-54de2bb71ad3)

Currently, the frontend is calling the `SaveChaosExperiment` mutation both on creation and update of an Experiment. This function creates a ChaosExperiment based on a manifest in yaml or json format submitted by the frontend. Previously, it generated the ChaosExperiment without caring whether the probe existed or not. 

With the update of Litmus to version 3, probe has become mandatory. Previously, we solved this by enforcing it on the frontend. However, in order for Litmus to provide external APIs in the future, we need to add logic to check for the existence of a probe on the backend as well. 

Therefore, if a probeRef does not exist during the process of parsing the manifest, we will call `probeService.AddProbe` directly to automatically create a probe. Once generated, we add its annotation via the `insertProbeRefAnnotation` function, create a ChaosExperiment, and return a 200 status.

Reply:
Does this code fit our logic? @S-ayanide 

```ts
/**
* If probeRef is not present in the manifest then return the fault name
* where the ref is missing otherwise return undefined
*/
const probeWithoutAnnotation = await (
  experimentHandler as KubernetesYamlService
)?.checkProbesInExperimentManifest(experiment?.manifest as KubernetesExperimentManifest);

/**
* Checks if probe metadata is already present in the manifest
*
* Returns true if probe metadata is present
*/
const doesProbeExists = await (experimentHandler as KubernetesYamlService)?.doesProbeMetadataExists(
  experiment?.manifest as KubernetesExperimentManifest
);

if (doesProbeExists) {
  showWarning(getString('probeMetadataExists'));
}

if (probeWithoutAnnotation && !doesProbeExists) {
  showError(`${getString('probeInFault')} ${probeWithoutAnnotation} ${getString('probeNotAttachedToRef')}`);
    return;
}
```

Reply:
Got it, I'm aligned with the above explanation and was thinking the same thing myself. The above logic is also correct. However, according to this statement above 

```
Therefore, if a probeRef does not exist during the process of parsing the manifest, we will call probeService.AddProbe directly to automatically create a probe. Once generated, we add its annotation via the insertProbeRefAnnotation function, create a ChaosExperiment, and return a 200 status.
```

I don't find the logic implemented in this PR where the `parseProbeFromManifest` backend handler is handling this change or the `insertProbeRefAnnotation` for example. I only see the probe service being added in this PR and being used.

Reply:
Found the change, the file was so big github squished it 😿